test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -0,0 +1,236 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import _sharded_tensor
+from torch.distributed._sharding_spec import ChunkShardingSpec
+from torch.testing._internal.common_distributed import (
+    MultiProcessTestCase,
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+
+
+class TestShardedTensor(MultiProcessTestCase):
+    @property
+    def world_size(self):
+        return 4
+
+    def init_pg(self):
+        dist.init_process_group(
+            backend="nccl",
+            world_size=self.world_size,
+            rank=self.rank,
+            init_method=f"file://{self.file_name}",
+        )
+
+    def setUp(self) -> None:
+        super().setUp()
+        self._spawn_processes()
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_chunked_sharding_basic(self):
+        self.init_pg()
+
+        for dim in [0, -2]:
+            spec = ChunkShardingSpec(
+                dim=dim,
+                placement=[
+                    "rank:0/cuda:0",
+                    "rank:1/cuda:1",
+                    "rank:2/cuda:2",
+                    "rank:3/cuda:3",
+                ],
+            )
+            sharded_tensor = _sharded_tensor.empty(spec, 10, 20)
+
+            # Validate local shard.
+            local_shards = sharded_tensor.local_shards()
+            self.assertEqual(1, len(local_shards))
+            local_shard = local_shards[0].shard
+            self.assertEqual(torch.device(f"cuda:{self.rank}"), local_shard.device)
+            if self.rank == 3:
+                self.assertEqual((1, 20), local_shard.size())
+            else:
+                self.assertEqual((3, 20), local_shard.size())
+
+            # Validate global metadata.
+            sharding_metadata = sharded_tensor.sharding_metadata()
+            self.assertEqual(4, len(sharding_metadata))
+
+            for rank, shard_metadata in enumerate(sharding_metadata):
+                self.assertEqual([rank * 3, 0], shard_metadata.shard_offsets)
+                if rank == 3:
+                    self.assertEqual([1, 20], shard_metadata.shard_lengths)
+                else:
+                    self.assertEqual([3, 20], shard_metadata.shard_lengths)
+                self.assertEqual(f'rank:{rank}/cuda:{rank}', shard_metadata.placement)
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_partial_world_size(self):
```

Comment:
Nit: Change "partial_world_size" to "sharded_by_2_of_4_devices"?

Reply:
It's up to you. Just feel that `partial_world_size` sounds less straightforward. If you keep this name, you can rename the above test case as "***_full_world_size".